src/main/java/picard/illumina/UnsortedBasecallsConverter.java

```diff
@@ -0,0 +1,95 @@
+package picard.illumina;
+
+import htsjdk.samtools.util.Log;
+import htsjdk.samtools.util.ProgressLogger;
+import htsjdk.samtools.util.SortingCollection;
+import picard.PicardException;
+import picard.illumina.parser.BaseIlluminaDataProvider;
+import picard.illumina.parser.ClusterData;
+import picard.illumina.parser.ReadStructure;
+import picard.illumina.parser.readers.BclQualityEvaluationStrategy;
+import picard.util.ThreadPoolExecutorUtil;
+import picard.util.ThreadPoolExecutorWithExceptions;
+
+import java.io.File;
+import java.time.Duration;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * UnortedBasecallsConverter utilizes an underlying IlluminaDataProvider to convert parsed and decoded sequencing data
+ * from standard Illumina formats to specific output records (FASTA records/SAM records). This data is processed
+ * on a tile by tile basis.
+ * <p>
+ * The underlying IlluminaDataProvider apply several optional transformations that can include EAMSS filtering,
+ * non-PF read filtering and quality score recoding using a BclQualityEvaluationStrategy.
+ * <p>
+ * The converter can also limit the scope of data that is converted from the data provider by setting the
+ * tile to start on (firstTile) and the total number of tiles to process (tileLimit).
+ * <p>
+ * Additionally, BasecallsConverter can optionally demultiplex reads by outputting barcode specific reads to
+ * their associated writers.
+ */
+public class UnsortedBasecallsConverter<CLUSTER_OUTPUT_RECORD> extends BasecallsConverter<CLUSTER_OUTPUT_RECORD> {
+    private static final Log log = Log.getInstance(UnsortedBasecallsConverter.class);
+    private final ProgressLogger readProgressLogger = new ProgressLogger(log, 1000000, "Read");
+    private final ProgressLogger writeProgressLogger = new ProgressLogger(log, 1000000, "Write");
```

Comment:
I'd argue we only need one progress logger - the fact that we may not write PF reads isn't that interesting.  Maybe just change to `Processed` and have one?